features/step_definitions/email_steps.rb

```diff
@@ -63,11 +63,26 @@
 end
 
 Then(/^#{capture_email} should contain "(.*)"$/) do |email_ref, text|
-  if email(email_ref).multipart?
-    email(email_ref).text_part.body.should =~ /#{text}/
-    email(email_ref).html_part.body.should =~ /#{text}/
+  mail = email(email_ref)
+
+  if mail.multipart?
+    [mail.text_part.body, mail.html_part.body].each do |part|
+      expect(part).to match(/#{text}/)
+    end
```

Comment:
I think it would be nicer to not loop in the two test steps because the body is short and there's only two things being iterated. Then it's a bit easier to see the difference between the multipart and non-multipart branches at a glance.